kit/Kit.cpp

```diff
@@ -2383,16 +2383,36 @@ void Document::drainQueue()
             }
         }
 
+        const size_t tilesQueued = _queue->getTileQueueSize();
         if (processInputEnabled() && !isLoadOngoing() &&
-            !isBackgroundSaveProcess() && _queue->getTileQueueSize() > 0)
+            !isBackgroundSaveProcess() && tilesQueued > 0)
         {
             std::vector<TileCombined> tileRequests = _queue->popWholeTileQueue();
-
-            // Put requests that include tiles in the visible area to the front to handle those first
-            std::partition(tileRequests.begin(), tileRequests.end(), [this](const TileCombined& req) {
-                return isTileRequestInsideVisibleArea(req); });
-            for (auto& tileCombined : tileRequests)
-                renderTiles(tileCombined);
+            std::vector<TileCombined*> earmarkedTiles;
+            size_t renderedTiles = 0;
+            for(TileCombined& tileCombined : tileRequests)
+            {
+                if (isIdle || (renderedTiles==0 && isTileRequestInsideVisibleArea(tileCombined)))
```

Comment:
If we are idle then render everything, that seems straight forward. But if we are not idle, and there are requests in the visible area, then we just render the first of those requests here? Why not just render all the visible area tiles?

Reply:
Additionally in non-idle mode, we would also render one invisible tile if all tiles are invisible.

In non-idle mode we assume to have limited time and only render one tile, then continue with the poll+drain loop.

Reply:
added comment